src/Markdig/Extensions/AutoLinks/AutoLinkOptions.cs

```diff
@@ -0,0 +1,22 @@
+namespace Markdig.Extensions.AutoLinks
+{
+    public class AutoLinkOptions
+    {
+        public AutoLinkOptions()
+        {
+            ValidPreviousCharacters = "*_~(";
+        }
+
+        public string ValidPreviousCharacters { get; set; }

```

Comment:
I'm a bit worried to introduce a mutable option class here, as a pipeline is supposed to be immutable after a build...
I don't recall if we have many cases already like this in the code base, but we should ideally strive for making these classes immutable

Reply:
Looking at other extensions, `PipeTable`, `Media` and `JiraLink` have mutable option classes while `SmartyPantOptions` is unmodifiable.

Would you rather these all have readonly fields and extra constructor params?

I can't really think of a clean way to enforce immutability otherwise (unless extensions made an internal copy and changes wouldn't propagate after the initial setup)

Reply:
yeah, C# is really poorly helping us here unfortunately... ok let it as it is for this one

We will have to later decide what to do with all these mutable classes and if we want to break them

Reply:
A property like https://github.com/lunet-io/markdig/pull/327/commits/befd1ca8464a7cc4ac5d0957f47ee726c8286c2d#diff-b59084d198ccb5dc29ab3f390ae3fcc7R15 falls under the same category them

Reply:
yes, as stated in the original issue #5 about immutable, these are open problems...